### PR TITLE
xcute: Implement blob rebuilder

### PR DESCRIPTION
**Pull request by AymericDu**
##### SUMMARY

Implement blob rebuilder

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

- `xcute`

##### SDS VERSION

```
openio 5.2.2
```

### CODE REVIEW

File: oio/cli/cluster/cluster.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def _list_services(self, parsed_args):
                 volume = tags.get('tag.vol', 'n/a')
                 service_id = tags.get('tag.service_id', 'n/a')
                 addr = srv['addr']
-                locked = boolean_value(tags.get('tag.lock', False))
+                locked = boolean_value(tags.get('tag.lock'), False)
```
Comment (loamhoof) [inline oio/cli/cluster/cluster.py]: Once again this has nothing to do in xcute?

File: oio/xcute/server.py [hunk 2]
```diff
@@ -73,10 +73,9 @@ def on_job_list(self, req):
             if job_class is None:
                 return HTTPBadRequest(UnknownJobTypeException.message)
 
-            job = job_class(self.conf, logger=self.logger)
-            job_config = data.get('config')
             # TODO: use lock
-            job_config, lock = job.sanitize_config(job_config)
+            job_config, lock = job_class.sanitize_config(
+                data.get('config') or dict())
```
Comment (loamhoof) [inline oio/xcute/server.py]: `data.get('config', dict)`?

File: oio/xcute/common/job.py [hunk 3]
```diff
@@ -39,7 +39,8 @@ def __init__(self, conf, logger=None):
         self.conf = conf
         self.logger = logger or get_logger(self.conf)
 
-    def sanitize_config(self, job_config):
+    @classmethod
+    def sanitize_config(cls, job_config):
```
Comment (loamhoof) [inline oio/xcute/common/job.py]: Can we change back `get_tasks` to be static as well?

File: oio/xcute/server.py (hunk 2 above)
Comment (AymericDu) [reply oio/xcute/server.py]:
```
>>> mydict = dict(mykey=None)
>>> mydict.get('mykey')
>>> mydict.get('mykey', dict())
>>> mydict.get('mykey') or dict()
{}
```

Comment (loamhoof) [reply oio/xcute/server.py]: Why would you have None though?

Comment (loamhoof) [reply oio/xcute/server.py]: If you can have None, you may as well check that you have the right type

File: oio/xcute/common/job.py (hunk 3 above)
Comment (AymericDu) [reply oio/xcute/common/job.py]: In all our code, we give the configuration and the logger in the constructor. So, I prefer to keep this same logic.

**Pull request merged**